adam/batch.py

```diff
@@ -140,16 +154,20 @@ def __init__(self, params):
         """
         # Make this a bit easier to get right by checking for parameters by unexpected
         # names.
-        supported_params = {
-            'epoch', 'state_vector', 'originator', 'object_name', 'object_id', 'mass',
-            'solar_rad_area', 'solar_rad_coeff', 'drag_area', 'drag_coeff',
-            'covariance', 'perturbation', 'hypercube'}
+        supported_params = {'epoch', 'state_vector', 'keplerian_elements', 'originator',
+                            'object_name', 'object_id', 'mass', 'solar_rad_area',
+                            'solar_rad_coeff', 'drag_area', 'drag_coeff', 'covariance',
+                            'perturbation', 'hypercube'}
         extra_params = params.keys() - supported_params
         if len(extra_params) > 0:
             raise KeyError("Unexpected parameters provided: %s" % (extra_params))
 
         self._epoch = params['epoch']  # Required.
-        self._state_vector = params['state_vector']  # Required.
+
+        if 'state_vector' not in params and 'keplerian_elements' not in params:
+            raise KeyError("Either state_vector or keplerian_elements must be provided.")
+        self._state_vector = params.get('state_vector') or [0, 0, 0, 0, 0, 0]  # Required in OPM.
```

Comment:
Maybe not specify it and add an if in the OPM generator instead?

Reply:
Ok, I have done so. I have mixed feelings about it. On the one hand, I see how it's weird to specify no state vector, then have something show up in the opm/get_state_vector. On the other hand, lots of other unspecified values have default values assigned here. On the third hand, if the point here is to isolate the user from the strange requirement of the opm that state vector always be specified, even if it will be ignored in favor of alternative coordinates, might as well do so thoroughly.